core/unix/src/TUnixSystem.cxx

```diff
@@ -568,12 +568,68 @@ TUnixSystem::~TUnixSystem()
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Initialize Unix system interface.
+extern "C"
+{
+    static int full_write(int fd, const char *text)
+    {
+      const char *buffer = text;
+      size_t count = strlen(text);
+      ssize_t written = 0;
+      while (count)
+      {
+        written = write(fd, buffer, count);
+        if (written == -1) 
+        {
+          if (errno == EINTR) {continue;}
+          else {return -errno;}
+        }
+        count -= written;
+        buffer += written;
+      }
+      return 0;
+    }
+
+    static int full_read(int fd, char *inbuf, size_t len)
```

Comment:
Add in an optional timeout to this function so the thread requesting the stack trace is less likely to timeout.